isb_lib/opencontext_adapter.py

```diff
@@ -113,21 +115,23 @@ def records_in_page(self):
                         self._date_start
                     )
                     data = {}
+                    self._past_date_start = True
                     break
                 # print(json.dumps(record, indent=2))
                 # raise NotImplementedError
                 yield record
                 num_records += 1
-            self.url = data.get("next-json")
-            if len(data.get("oc-api:has-results", {})) < _page_size:
-                more_work = False
-            elif self.url is None:
-                # If we hit the last page, there won't be a 'next-json' key
-                more_work = False
-            elif 0 < self._max_entries <= num_records:
-                more_work = False
-            elif num_records == self._page_size:
+
+            if (
+                len(data.get("oc-api:has-results", {})) < _page_size
+                or self.url is None
+                or 0 < self._max_entries <= num_records
+                or num_records == self._page_size
+                or self._past_date_start
+            ):
```

Comment:
This was just cleanup as the if-statement kept growing and it had the same 1-line statement in 4 places.